kivakit-microservice/src/main/java/com/telenav/kivakit/microservice/internal/protocols/rest/plugins/jetty/cycle/JettyMicroserviceResponse.java

```diff
@@ -168,23 +169,54 @@ public Version version()
      */
     public void writeObject(MicroservletResponse response)
     {
+        var responseWritten = false;
+
         // Validate the response
         if (response != null)
         {
             // and if the response is invalid (any problems go into the response object),
-            if (!response.isValid(response))
+            if (response.isValid(response))
+            {
+                var responseType = cycle.request().parameters().getOrDefault("response-type", "always-okay");
+
+                switch (responseType)
+                {
+                    case "http-status":
+                        var json = errors.isEmpty()
+                                ? toJson(response)
+                                : toJson(errors);
+                        writeResponse(json);
+                        httpStatus(errors.httpStatus());
+                        responseWritten = true;
+                        break;
+
+                    case "always-okay":
+                        writeResponse(toJson(response));
+                        writeResponse(toJson(errors));
+                        httpStatus(HttpStatus.OK);
+                        responseWritten = true;
+                        break;
+
+                    default:
+                        problem(HttpStatus.BAD_REQUEST, "Response-type must be 'http-status', or 'always-okay', if not omitted");
+                        break;
+                }
+            }
+            else
             {
                 // then transmit a problem message.
-                problem("Response object is invalid");
+                problem(HttpStatus.INTERNAL_SERVER_ERROR, "Internal error: Response object is invalid");
             }
         }
+        else
+        {
+            problem(HttpStatus.INTERNAL_SERVER_ERROR, "Internal error: Response object is invalid");
```

Comment:
The text of the response should be different in the case of `null` vs. `!response.isValid()` or it will make differentiating the two cases from a log record much harder.